src/apis/token_api.ts

```diff
@@ -173,11 +176,23 @@ export class TokenAPI {
 
         await this.assert.token.implementsERC20(tokenContract);
 
-        // We set an allowance to be "unlimited" by setting it to
-        // it's maximum possible value -- namely, 2^256 - 1.
-        const unlimitedAllowance = new BigNumber(2).pow(256).sub(1);
+        return this.setProxyAllowanceAsync(tokenAddress, UNLIMITED_ALLOWANCE, options);
+    }
+
+    /**
+     * Eventually determines whether the allowance for the specified owner is unlimited.
+     *
+     * @param  tokenAddress address of the ERC20 token.
+     * @param  ownerAddress the owner whose allowance is being queried.
+     * @returns {Promise<boolean>}
+     */
+    public async hasUnlimitedAllowance(
+        tokenAddress: string,
+        ownerAddress: string,
+    ): Promise<boolean> {
+        const existingAllowance = await this.getProxyAllowanceAsync(tokenAddress, ownerAddress);
 
-        return this.setProxyAllowanceAsync(tokenAddress, unlimitedAllowance, options);
+        return existingAllowance.greaterThanOrEqualTo(UNLIMITED_ALLOWANCE);
```

Comment:
one thing you could do here is extract this, e.g.:

```javascript
isUnlimited(amount) {
  return amount.gte(UNLIMITED_AMOUNT);
}
```

This would allow you to check, later on, if a given amount that you have is the unlimited amount.

```javascript
resolve({
  allowance: allowanceAmount,
  hasUnlimitedAllowance: dharma.token.isUnlimitedAllowance(allowanceAmount),
});
```

Hence saving a query.

Reply:
Great call